cli/src/lib.rs

```diff
@@ -512,32 +512,70 @@ fn override_toolchain(cfg_override: &ConfigOverride) -> Result<RestoreToolchainC
 
             Ok(version)
         }
+        // Function to check the installed solana version
+        fn is_solana_version_installed(version: &str) -> Result<bool> {
+            let output = std::process::Command::new("solana")
+                .arg("--version")
+                .output()?;
+            let output_version = std::str::from_utf8(&output.stdout)?;
+            Ok(output_version.contains(version))
+        }
```

Comment:
Please check out https://github.com/coral-xyz/anchor/pull/2757#discussion_r1440086674, this function doesn't do what it's supposed to do.

Reply:
> Please check out [#2757 (comment)](https://github.com/coral-xyz/anchor/pull/2757#discussion_r1440086674), this function doesn't do what it's supposed to do.

That's what we wanna do right? like check whether the installed version is the current version or not?
or we want to check if the given version is installed regardless of whether it is the current version?

Reply:
> That's what we wanna do right? like check whether the installed version is the current version or not? or we want to check if the given version is installed regardless of whether it is the current version?

With `solana-install`, we can install multiple versions in the system. So I think you need to check whether the version that Anchor want is already installed or not.

When we run `solana-install init <VERSION>` two scenarios can occur:
1. already installed -> switch to `<VERSION>`
2. not already installed -> install it -> switch to `<VERSION>`

I can be wrong but this is want I understand. hope this help (and correct 😂)

Reply:
@acheroncrypto this is what you meant?

Reply:
if this what we want then i think this implementation will solve that

```rs
fn is_solana_version_installed(version: &str) -> Result<bool> {
    let output = std::process::Command::new("solana")
        .arg("--version")
        .output()?;

    if output.status.success() {
        let output_version = std::str::from_utf8(&output.stdout)?;
        Ok(output_version.lines().any(|line| line.contains(version)))
    } else {
        Err(anyhow!("Failed to run 'solana --version'"))
    }
}
```

Reply:
@qapquiz is right(https://github.com/coral-xyz/anchor/pull/2757#discussion_r1442734619).

> if this what we want then i think this implementation will solve that

Need to get all installed versions but running `solana --version` like in the code you've shared only gets you the current version.

Reply:
Okay then need to check the list like i shared above snippet, right?